File: MainController.java
package com.example.demo;

import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Node;
import javafx.scene.chart.PieChart;
import javafx.scene.control.*;
import javafx.scene.layout.BorderPane;
import javafx.scene.text.Font;

import javax.swing.*;
import java.io.IOException;
import java.util.HashMap;


public class MainController {
    //Declares controllers
    private ForceMetricsController forceMetricsController;
    private ForceParametersController forceParametersController;
    private ForceCostController forceCostController;
    private AssetController assetController;
    private PurchasingController purchasingController;
    @FXML
    private BorderPane mainBorderPane;
    @FXML
    private Label titleLabel;
    //Loads FXML and controllers for child panes
    private Node loadFXML(String filename) {
        try {
            FXMLLoader loader = new FXMLLoader(getClass().getResource(filename));
            Node node = loader.load();
            switch(filename) {
                case "ForceMetrics.fxml":
                    forceMetricsController = loader.getController();
                    break;
                case "ForceParameters.fxml":
                    forceParametersController = loader.getController();
                    break;
                case "ForceCost.fxml":
                    forceCostController = loader.getController();
                    break;
                case "Asset.fxml":
                    assetController = loader.getController();
                    break;
                case "Purchasing.fxml":
                    purchasingController = loader.getController();
                    break;
            }
            return node;
        } catch (IOException e) {
            e.printStackTrace();
            System.exit(1);
            return null;
        }
    }
    private Node titleScreenView = loadFXML("TitleScreen.fxml");
    private Node forceMetricsView = loadFXML("ForceMetrics.fxml");
    private Node forceParametersView = loadFXML("ForceParameters.fxml");
    private Node forceCostView = loadFXML("ForceCost.fxml");
    private Node assetView = loadFXML("Asset.fxml");
    private Node purchasingView = loadFXML("Purchasing.fxml");


    @FXML
    private void onPurchasingClick (ActionEvent e) {
        purchasingController.populate();
        purchasingController.alert();
        mainBorderPane.setCenter(purchasingView);
        titleLabel.setText("Purchasing");

    }
    @FXML
    private void onForceParametersClick(ActionEvent e) {
        forceParametersController.populate();
        mainBorderPane.setCenter(forceParametersView);
        titleLabel.setText("Force Parameters");
    }
    @FXML
    private void onForceCostClick (ActionEvent e) {
        forceCostController.populate();
        mainBorderPane.setCenter(forceCostView);
        titleLabel.setText("Force Cost");
    }
    @FXML private void onForceMetricsClick (ActionEvent e) {
        forceMetricsController.populate();
        mainBorderPane.setCenter(forceMetricsView);
        titleLabel.setText("Force Metrics");
    }
    @FXML
    private void onMQ4CClick (ActionEvent e) {
        assetController.populate(PurchasingBackend.cart.MQ4C);
        mainBorderPane.setCenter(assetView);
        titleLabel.setText("Northrop Grumman MQ-4C Triton");
    }
    @FXML
    private void onMQ9BClick (ActionEvent e) {
        assetController.populate(PurchasingBackend.cart.MQ9B);
        mainBorderPane.setCenter(assetView);
        titleLabel.setText("GA Aeronautics MQ-9B SeaGuardian");
    }
    @FXML
    private void onP1Click (ActionEvent e) {
        assetController.populate(PurchasingBackend.cart.P1);
        mainBorderPane.setCenter(assetView);
        titleLabel.setText("Kawasaki P-1");
    }
    @FXML
    private void onIntegratorClick (ActionEvent e) {
        assetController.populate(PurchasingBackend.cart.integrator);
        mainBorderPane.setCenter(assetView);
        titleLabel.setText("Boeing Insitu Integrator ER");
    }
    @FXML
    private void onRQ58Click (ActionEvent e) {
        assetController.populate(PurchasingBackend.cart.RQ58);
        mainBorderPane.setCenter(assetView);
        titleLabel.setText("Kratos RQ-58 Valkyrie");
    }
    @FXML
    private void onThunderheadClick (ActionEvent e) {
        assetController.populate(PurchasingBackend.cart.thunderhead);
        mainBorderPane.setCenter(assetView);
        titleLabel.setText("Raven Aerostar Thunderhead");
    }
    //Basic utilities
    public static int getInt(Label field) {
        int num = 0;
        try {
            num = Integer.parseInt(field.getText());
        } catch (Exception e) {
            Alert alert = new Alert(Alert.AlertType.CONFIRMATION, "Please ensure the text you entered is an integer.", ButtonType.YES, ButtonType.NO, ButtonType.CANCEL);
            alert.showAndWait();
            return num;
        }
        return num;
    }
    public static int getInt(TextField field) {
        int num = 0;
        try {
            num = Integer.parseInt(field.getText());
        } catch (Exception e) {
            Alert alert = new Alert(Alert.AlertType.CONFIRMATION, "Please ensure the text you entered is an integer.", ButtonType.YES, ButtonType.NO, ButtonType.CANCEL);
            alert.showAndWait();
            return num;
        }
        return num;
    }
    public void initialize() {
        mainBorderPane.setCenter(titleScreenView);
        titleLabel.setText("Welcome");
    }
}

File: main_controller.py
import sys
from PyQt5.QtWidgets import QApplication, QLabel, QMainWindow, QMessageBox, QVBoxLayout, QWidget, QLineEdit
from PyQt5.uic import loadUi

class MainController(QMainWindow):
    def __init__(self):
        super(MainController, self).__init__()
        self.forceMetricsController = None
        self.forceParametersController = None
        self.forceCostController = None
        self.assetController = None
        self.purchasingController = None

        self.mainBorderPane = QWidget()
        self.titleLabel = QLabel("Welcome")
        self.titleScreenView = self.load_ui("TitleScreen.ui")
        self.forceMetricsView = self.load_ui("ForceMetrics.ui")
        self.forceParametersView = self.load_ui("ForceParameters.ui")
        self.forceCostView = self.load_ui("ForceCost.ui")
        self.assetView = self.load_ui("Asset.ui")
        self.purchasingView = self.load_ui("Purchasing.ui")

        layout = QVBoxLayout()
        layout.addWidget(self.titleLabel)
        layout.addWidget(self.mainBorderPane)
        self.setLayout(layout)

        self.initialize()

    def load_ui(self, filename):
        try:
            widget = loadUi(filename)
            if filename == "ForceMetrics.ui":
                self.forceMetricsController = widget
            elif filename == "ForceParameters.ui":
                self.forceParametersController = widget
            elif filename == "ForceCost.ui":
                self.forceCostController = widget
            elif filename == "Asset.ui":
                self.assetController = widget
            elif filename == "Purchasing.ui":
                self.purchasingController = widget
            return widget
        except Exception as e:
            print(e)
            sys.exit(1)

    def on_purchasing_click(self):
        self.purchasingController.populate()
        self.purchasingController.alert()
        self.mainBorderPane.setCentralWidget(self.purchasingView)
        self.titleLabel.setText("Purchasing")

    def on_force_parameters_click(self):
        self.forceParametersController.populate()
        self.mainBorderPane.setCentralWidget(self.forceParametersView)
        self.titleLabel.setText("Force Parameters")

    def on_force_cost_click(self):
        self.forceCostController.populate()
        self.mainBorderPane.setCentralWidget(self.forceCostView)
        self.titleLabel.setText("Force Cost")

    def on_force_metrics_click(self):
        self.forceMetricsController.populate()
        self.mainBorderPane.setCentralWidget(self.forceMetricsView)
        self.titleLabel.setText("Force Metrics")

    def on_mq4c_click(self):
        self.assetController.populate(PurchasingBackend.cart.MQ4C)
        self.mainBorderPane.setCentralWidget(self.assetView)
        self.titleLabel.setText("Northrop Grumman MQ-4C Triton")

    def on_mq9b_click(self):
        self.assetController.populate(PurchasingBackend.cart.MQ9B)
        self.mainBorderPane.setCentralWidget(self.assetView)
        self.titleLabel.setText("GA Aeronautics MQ-9B SeaGuardian")

    def on_p1_click(self):
        self.assetController.populate(PurchasingBackend.cart.P1)
        self.mainBorderPane.setCentralWidget(self.assetView)
        self.titleLabel.setText("Kawasaki P-1")

    def on_integrator_click(self):
        self.assetController.populate(PurchasingBackend.cart.integrator)
        self.mainBorderPane.setCentralWidget(self.assetView)
        self.titleLabel.setText("Boeing Insitu Integrator ER")

    def on_rq58_click(self):
        self.assetController.populate(PurchasingBackend.cart.RQ58)
        self.mainBorderPane.setCentralWidget(self.assetView)
        self.titleLabel.setText("Kratos RQ-58 Valkyrie")

    def on_thunderhead_click(self):
        self.assetController.populate(PurchasingBackend.cart.thunderhead)
        self.mainBorderPane.setCentralWidget(self.assetView)
        self.titleLabel.setText("Raven Aerostar Thunderhead")

    @staticmethod
    def get_int(field):
        try:
            return int(field.text())
        except ValueError:
            msg = QMessageBox()
            msg.setIcon(QMessageBox.Warning)
            msg.setText("Please ensure the text you entered is an integer.")
            msg.setStandardButtons(QMessageBox.Ok)
            msg.exec_()
            return 0

    def initialize(self):
        self.mainBorderPane.setCentralWidget(self.titleScreenView)
        self.titleLabel.setText("Welcome")

if __name__ == "__main__":
    app = QApplication(sys.argv)
    mainController = MainController()
    mainController.show()
    sys.exit(app.exec_())
